rocky/rocky/views/ooi_report.py

```diff
@@ -198,47 +243,54 @@ def get(self, request, *args, **kwargs):
-        report_data["ooi"] = get_ooi_dict(self.ooi)
+        return self.get_report(
+            self.ooi.object_type,
+            self.ooi.human_readable,
+            self.tree.store,
+            self._generate_report_file_name(),
+            get_ooi_url("ooi_report", self.ooi.primary_key, self.organization.code),
+        )
 
-        # request pdf from keiko
-        try:
-            report_id = keiko_client.generate_report("bevindingenrapport", report_data, "dutch.hiero.csv")
-        except HTTPError as e:
-            messages.error(self.request, _("Error generating report: {}").format(e))
-            return redirect(get_ooi_url("ooi_report", self.ooi.primary_key, self.organization.code))
-
-        # generate file name
-        report_name = "bevindingenrapport"
-        org_code = self.organization.code
-        ooi_id = self.ooi.primary_key
-        valid_time = self.get_observed_at().isoformat()
-        current_time = datetime.now(timezone.utc).isoformat()
-        language = "nl"
+    def _generate_report_file_name(self):
         report_file_name = "_".join(
             [
-                report_name,
-                language,
-                org_code,
-                ooi_id,
-                valid_time,
-                current_time,
+                "bevindingenrapport",
+                "nl",
+                self.organization.code,
+                self.ooi.primary_key,
+                self.get_observed_at().isoformat(),
+                datetime.now(timezone.utc).isoformat(),
             ]
         )
         # allow alphanumeric characters, dashes and underscores, replace rest with underscores
         report_file_name = re.sub("[^0-9a-zA-Z-]", "_", report_file_name)
         report_file_name = f"{report_file_name}.pdf"
 
-        # open pdf as attachment
-        try:
-            return FileResponse(keiko_client.get_report(report_id), as_attachment=True, filename=report_file_name)
-        except (HTTPError, ReportNotFoundException):
-            messages.error(
-                self.request, _("Error generating report: Timeout reached. See Keiko logs for more information.")
-            )
-            return redirect(get_ooi_url("ooi_report", self.ooi.primary_key, self.organization.code))
+        return report_file_name
+
+
+@class_view_decorator(otp_required)
+class FindingReportPDFView(FindingListView, ReportPDFView):
+    paginate_by = None
+
+    def get(self, request, *args, **kwargs):
+        findings = [item["finding"] for item in super().get_queryset()]
+
+        return self.get_report(
+            "Organization",
+            self.organization.name,
+            {finding.primary_key: finding for finding in findings},  # Create store from finding list
+            self._generate_report_file_name(),
+            reverse("finding_list", kwargs={"organization_code": self.organization.code}),
+        )
+
+    def _generate_report_file_name(self):
```

Comment:
I was wondering if this replacement is necessary since neither of the joined elements contain these characters (except for the colon used in the datetime's format, but that doesn't necessarily justify a regex replacement since you can format your own datetime string)

Reply:
Not sure to be honest, but perhaps primary keys of ooi's could give some issues?

Reply:
Ah that's the other one. But organization codes can contain hyphens perhaps?

Reply:
Made some updates for this one :+1: